homie5/formats/integer.yml

```
@@ -1,5 +1,107 @@
 description: validating integer formats
 
-# TODO: implement these tests
+tests:
+  - description: proper open-ended minimum format is accepted
+    testtype: propertydescription
+    definition:
+      datatype: integer
+      format: "0:"
+    valid: true
 
-tests: []
+  - description: proper open-ended maximum format is accepted
+    testtype: propertydescription
+    definition:
+      datatype: integer
+      format: ":100"
+    valid: true
+
+  - description: proper closed range format is accepted
+    testtype: propertydescription
+    definition:
+      datatype: integer
+      format: "10:50"
+    valid: true
+
+  - description: proper format with step value is accepted
+    testtype: propertydescription
+    definition:
+      datatype: integer
+      format: "0:100:5"
+    valid: true
+
+  - description: negative range with step value is accepted
+    testtype: propertydescription
+    definition:
+      datatype: integer
+      format: "-100:0:10"
+    valid: true
+
+  - description: single point range with step value is accepted
+    testtype: propertydescription
+    definition:
+      datatype: integer
+      format: "5:5:1"
+    valid: true
+
+  - description: reverse range is rejected
+    testtype: propertydescription
+    definition:
+      datatype: integer
+      format: "50:10"
+    valid: false
+
+  - description: step value of zero is rejected
+    testtype: propertydescription
+    definition:
+      datatype: integer
+      format: "0:100:0"
+    valid: false
+
+  - description: step value larger than range is rejected
+    testtype: propertydescription
+    definition:
+      datatype: integer
+      format: "0:10:20"
+    valid: false
```

```suggestion
  - description: step value larger than range is accepted
    testtype: propertydescription
    definition:
      datatype: integer
      format: "0:10:20"
    valid: true
```
collides with `single point range with step value is accepted` above